sdk/storage/Azure.Storage.Blobs/tests/MockKeyEncryptionKey.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Security.Cryptography;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Cryptography;
+using Microsoft.Azure.KeyVault.Core;
+
+namespace Azure.Storage.Tests.Shared
+{
+    /// <summary>
+    /// Mock for a key encryption key. Not meant for production use.
+    /// </summary>
+    internal class MockKeyEncryptionKey : IKeyEncryptionKey, IKeyEncryptionKeyResolver,
+        IKey, IKeyResolver // for track 1 compatibility tests
```

Comment:
I'd rather break it down into 4 separate types. Unless we have a good reason to create [God object](https://en.wikipedia.org/wiki/God_object)

Reply:
It makes cleaner tests. This improved my test-writing ability substantially.

I wouldn't consider this a god class, as it's really only doing two small actions and exposing them repeatedly through a large variety of interfaces understood by track 1 and track 2. This ensures we are doing the same operations across track 1 and 2 in our testing so we can know that we're emulating comparable situations.

Reply:
Then the class name is wrong (and honestly I'm not sure how to name it, which is another hint this should be spitted). I'm not sure how test-writing ability is improved by combining loosely related functionalities - if it's creating 4 fields/vars in test vs 1 then I think it's not a big improvement (ink is cheap on the computer).
I've seen test aids like that in the past (that were trying to abstract too much). It usually ended up not well in long run. (coupling functionality is usually more expensive to change in long ran than not having to write extra variable names).
Not a very big deal though at this point, we can break it later when we feel more pain. Make your call here.

Reply:
We discussed elsewhere to make better use of Moq anyway. I'll go that route, having functions that do the work being placed onto different mocks. Will accomplish my goal of maintaining code to do these mock operations in one place.